tracker-edtracker/ftnoir_tracker_edtracker_dialog.cpp

```diff
@@ -0,0 +1,65 @@
+#include "ftnoir_tracker_edtracker.h"
+#include "api/plugin-api.hpp"
+
+dialog_edtracker::dialog_edtracker() {
+
+    ui.setupUi(this);
+
+    connect(ui.buttonBox, SIGNAL(accepted()), this, SLOT(doOK()));
+    connect(ui.buttonBox, SIGNAL(rejected()), this, SLOT(doCancel()));
+
+    updateDevices();
+
+    tie_setting(s.idx_x, ui.input_x);
+    tie_setting(s.idx_y, ui.input_y);
+    tie_setting(s.idx_z, ui.input_z);
+
+    tie_setting(s.add_yaw, ui.add_yaw);
+    tie_setting(s.add_pitch, ui.add_pitch);
+    tie_setting(s.add_roll, ui.add_roll);
+
+}
+
+void dialog_edtracker::doOK() {
+    int idx = ui.dev->currentIndex();
+    static const joys def { {}, {} };
+    auto val = joys_.value(idx, def);
+    s.dev = val.device;
+    s.b->save();
+    close();
+}
+
+void dialog_edtracker::doCancel() {
+    close();
+}
+
+void dialog_edtracker::updateDevices() {
+    getJoysticks();
```

Comment:
Don't use a static variable due to lack of thread safety (opentrack can run as a library).